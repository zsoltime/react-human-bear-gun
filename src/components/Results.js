import React from 'react';
import PropTypes from 'prop-types';

import Icon from './Icons';
import { stringOrNull } from '../helpers';

const Results = ({
  computerChoice,
  userChoice,
  points,
  result,
  winner,
}) => (
  <div className="results" aria-atomic="true" aria-live="polite">
    <p
      className={
        winner
          ? winner === 'user'
            ? 'result result--winner'
            : 'result result--loser'
          : 'result'
      }
    >
      <Icon
        aria-hidden="true"
        className={
          userChoice
            ? 'icon icon--small'
            : 'icon icon--small icon--thinking--mirrored'
        }
        focusable="false"
        type={userChoice || 'Think'}
      />
      <span className="visuallyhidden">
        {userChoice
          ? `You chose ${userChoice}`
          : "You're still thinking"}
      </span>
    </p>
    <p
      className={
        winner
          ? winner === 'computer'
            ? 'result result--winner'
            : 'result result--loser'
          : 'result'
      }
    >
      <Icon
        aria-hidden="true"
        className={
          computerChoice
            ? 'icon icon--small'
            : 'icon icon--small icon--thinking'
        }
        focusable="false"
        type={computerChoice || 'Think'}
      />
      <span className="visuallyhidden">
        {computerChoice
          ? `Computer chose ${computerChoice}`
          : 'Computer is still thinking'}
      </span>
    </p>
    <p aria-hidden="true" className="result result--small">
      You: ({points.user})
    </p>
    <p aria-hidden="true" className="result">
      vs
    </p>
    <p aria-hidden="true" className="result result--small">
      Computer: ({points.computer})
    </p>
    {result && <p className="result">{result}</p>}
  </div>
);

Results.propTypes = {
  computerChoice: stringOrNull,
  userChoice: stringOrNull,
  points: PropTypes.shape({
    computer: PropTypes.number.isRequired,
    user: PropTypes.number.isRequired,
  }),
  result: stringOrNull,
  winner: stringOrNull,
};

export default Results;
