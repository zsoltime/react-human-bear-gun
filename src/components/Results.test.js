import React from 'react';
import { shallow } from 'enzyme';

import Results from './Results';

const CHOICES = ['Human', 'Bear', 'Gun'];

describe('<Results />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Results
        computerChoice={CHOICES[0]}
        userChoice={CHOICES[1]}
        points={{ computer: 3, user: 5 }}
        result="Result text"
        winner="user"
      />
    );

    expect(wrapper.find('.results')).toMatchSnapshot();
  });
});
