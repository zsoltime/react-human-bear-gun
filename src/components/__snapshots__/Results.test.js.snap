// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> matches snapshot 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  className="results"
>
  <p
    className="result result--winner"
  >
    <Icon
      aria-hidden="true"
      className="icon icon--small"
      focusable="false"
      type="Bear"
    />
    <span
      className="visuallyhidden"
    >
      You chose Bear
    </span>
  </p>
  <p
    className="result result--loser"
  >
    <Icon
      aria-hidden="true"
      className="icon icon--small"
      focusable="false"
      type="Human"
    />
    <span
      className="visuallyhidden"
    >
      Computer chose Human
    </span>
  </p>
  <p
    aria-hidden="true"
    className="result result--small"
  >
    You: (
    5
    )
  </p>
  <p
    aria-hidden="true"
    className="result"
  >
    vs
  </p>
  <p
    aria-hidden="true"
    className="result result--small"
  >
    Computer: (
    3
    )
  </p>
  <p
    className="result"
  >
    Result text
  </p>
</div>
`;
